### PR TITLE
add solution

### PR DESCRIPTION
**Pull request by vitvadym**

### CODE REVIEW

**Review by silchyn — APPROVED** [review]

**Review by silchyn — CHANGES_REQUESTED** [review]
checklist rule `#4`

File: src/transformState.js [hunk 1]
```diff
@@ -5,7 +5,37 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  const currentState = state;
```
Comment (33Orange) [inline src/transformState.js]: It is unnecessary to create a new variable 'currentState'. You can directly modify 'state' without creating a new reference.

File: src/transformState.js [hunk 2]
```diff
@@ -5,7 +5,37 @@
...
+        }
+
+        break;
+
+      default:
+        throw Error(`Wrong action: ${action.type}`);
+    }
+  }
+
+  return state;
```
Comment (33Orange) [inline src/transformState.js]: Returning 'state' is not necessary, as the function is supposed to modify the original object. Remove the 'return' statement.

File: src/transformState.js [hunk 3]
```diff
@@ -5,7 +5,37 @@
...
+  const currentState = state;
+
+  for (const action of actions) {
+    const { type, keysToRemove, extraData } = action;
+
+    switch (type) {
+      case 'clear':
+        for (const key in currentState) {
+          delete currentState[key];
+        };
```
Comment (33Orange) [inline src/transformState.js]: Let's move this code into a separate function after `transformState`

File: src/transformState.js [hunk 4]
```diff
@@ -5,7 +5,37 @@
...
+
+      case 'addProperties':
+        Object.assign(currentState, extraData);
+
+        break;
+
+      case 'removeProperties':
+        for (const keyToRemove of keysToRemove) {
+          delete currentState[keyToRemove];
+        }
```
Comment (33Orange) [inline src/transformState.js]: Let's move this code into a separate function after `transformState`

**Review by 33Orange — CHANGES_REQUESTED** [review]
Good job.

A few moments to fix

**Review by naziamoff — APPROVED** [review]
great job 👍 
If we create a helper function - we usually move it to a separate file. You can practice this if you want

**Review by silchyn — APPROVED** [review]

**Pull request closed without merging**